core/templates/dev/head/pages/exploration_player/AnswerClassificationService.js

```
@@ -151,6 +153,63 @@ oppia.factory('AnswerClassificationService', [
         }
         return deferred.promise;
       }
+
+      /**
+       * Gets the matching answer group.
+       *
+       * @param {string} explorationId - The exploration ID.
+       * @param {object} oldState - The state where the user submitted the
+       *   answer.
+       * @param {*} answer - The answer that the user has submitted.
+       * @param {boolean} isInEditorMode - Whether the function is being called
+       *   in editor mode.
+       * @param {function} interactionRulesService - The service which contains
+       *   the explicit rules of that interaction.
+       *
+       * @return {object} An object representing the answer group with the
+       *     following properties:
+       * <ul>
+       *   <li> **outcome**: the outcome of the answer group
+       *   <li> **answerGroupIndex**: the index of the matched answer group
+       *   <li> **ruleIndex**: the index of the rule in the matched answer
+       *            group
+       * </ul>
+       */
+      getMatchingClassificationResultNew: function(
+          explorationId, oldState, answer, isInEditorMode,
+          interactionRulesService) {
+        var result = null;
+        var answerGroups = oldState.interaction.answerGroups;
+        var defaultOutcome = oldState.interaction.defaultOutcome;
+        if (interactionRulesService) {
+          result = classifyAnswer(
+            answer, answerGroups, defaultOutcome, interactionRulesService);
+        } else {
+          alertsService.addWarning(
+            'Something went wrong with the exploration: no ' +
+            'interactionRulesService was available.');
+          // @prasanna08 @seanlip. Not sure about this part.
```

@pranavsid98 This is the case when there is no explicit rules service associated with the interaction, which seems like an error. What is your specific question (and best guess at a solution), though? Without knowing that, it's hard to know how to respond.

---

@seanlip If we are going with async method right now, we can return deferred.reject(). But, when we do eventually make it a synchronous process, this should probably raise the exception and just end the execution there. I assume we can't do an empty return. What would be a good equivalent for deferred.reject()?

---

throw Error(...)?

---

I mean, in normal operation, this shouldn't happen...

---

@seanlip We don't need to throw error do we? Doesn't the alertService add the warning for us already?

---

That's the user-visible version. But you'll need to throw an error to avoid further code execution, I think.

---

Makes sense. But since we aren't really focusing on changing this into a synchronous method right now, we can let it be for now. We can make a separate PR as you suggested to achieve the above.

---

Yep, sounds good!